components/supervisor/pkg/supervisor/supervisor.go

```diff
@@ -1106,6 +1109,16 @@ func buildChildProcEnv(cfg *Config, envvars []string, runGP bool, setJavaXmx boo
 	envs["HOME"] = "/home/gitpod"
 	envs["USER"] = "gitpod"
 
+	if cpuCount, ok := envs["GITPOD_CPU_COUNT"]; ok && !isSetJavaProcessorCount {
+		if _, exists := envs["JAVA_TOOL_OPTIONS"]; exists {
```

Comment:
fyi intellij overrides it to avoid showing a warning to users each time, one has to configure it via jvmoptions for them

Reply:
see https://github.com/gitpod-io/gitpod/blob/6e8dd4e2b9ebe0f12602a13ca1b14017a4676c9b/components/ide/jetbrains/launcher/main.go#L647-L650

Reply:
Maybe we could parse given JAVA_TOOL_OPTIONS and append them to idea64.vmoptions automatically, we still should be careful that we dont reduce xmx less than 2048m